src/model.cpp

```diff
@@ -219,4 +222,145 @@ size_t Model::unitsCount() const
     return mPimpl->mUnits.size();
 }
 
+typedef std::shared_ptr<ImportedEntity> ImportedEntityPtr;
+
+/**
+ * @brief Resolve the path of the given filename using the given base.
+ *
+ * Resolves the full path to the given @p filename using the @p base.
+ *
+ * This function is only intended to work with local files.  It may not
+ * work with bases that use the 'file://' prefix.
+ *
+ * @param filename The @c std::string relative path from the base path.
+ * @param base The @c std::string location on local disk for determining the full path from.
+ * @return The full path from the @p base location to the @p filename
+ */
+std::string resolvePath(const std::string& filename, const std::string& base)
+{
+    // we can be naive here as we know what we are dealing with
+    std::string path = base.substr(0, base.find_last_of('/')+1) + filename;
+    return path;
+}
+
+void resolveImport(ImportedEntityPtr importedEntity,
+                   const std::string& baseFile)
+{
+    if (importedEntity->isImport()) {
+        libcellml::ImportSourcePtr importSource = importedEntity->getImportSource();
+        if (!importSource->hasModel()) {
+            std::string url = resolvePath(importSource->getUrl(), baseFile);
+            std::ifstream t(url);
+            if (t.good()) {
+                std::stringstream buffer;
+                buffer << t.rdbuf();
+                libcellml::Parser p;
+                libcellml::ModelPtr model = p.parseModel(buffer.str());
+                if (model) {
```

Comment:
From what I can see in `Parser::parseModel()`, we will always have a model, so no point in testing for it?